app/models/custom_field.rb

```diff
@@ -70,6 +72,10 @@ def self.loan_questions(field_set = nil)
       { internal_name: field_set })
   end
 
+  def self.sort_by_required(loan)
+    all.sort_by { |i| [i.required_for?(loan) ? 0 : 1, i.position] }
```

Comment:
Curious why this can't be a normal scope. Hard to tell since `required_for?` is not implemented.